content/src/draw-protractor.js

```diff
@@ -17,8 +17,24 @@ function to360(v) {
 
 function renderProtractor(canvas, step) {
   var ctx = canvas[0].getContext('2d');
-  ctx.resetTransform();
+  ctx.setTransform(1, 0, 0, 1, 0, 0);
   ctx.clearRect(0, 0, canvas.width(), canvas.height());
+  if (step.startCoords != null) {
+    t =  new $.turtle.Turtle;
+    t.appendTo('.preview')
+    var offset = $('.preview').offset()
+    t.moveto({pageX: step.startCoords.pageX + offset.left, pageY:step.startCoords.pageY + offset.top});
+    t.wear('gray');
+    if (step.endCoords != null) {
+      if (step.command == "rt") {
+        t.rt(step.endCoords.direction);
+      } 
```

Comment:
Since step.command can't be both rt and lt, maybe use else if? Or maybe switch. For multiple cases, probably use switch. For two, use whichever you prefer.